### PR TITLE
Replace errors_on/2 with errors_on/1

### PR DESCRIPTION
**Pull request by wojtekmach**
As discussed on https://github.com/phoenixframework/phoenix/commit/d77cddbb0dc00d6f1edcbcb21b7f266a3d0614a2#commitcomment-21075635

### CODE REVIEW

**Comment by wojtekmach** [comment]
Another idea is to actually have `errors_on/1` instead, that returns the map. We'd support the original use case, but also have an ability to inspect all errors:

```elixir
assert "password is too short" in errors_on(changeset).password
assert %{password: ["password is too short"]} = errors_on(changeset)
```

File: installer/templates/phx_ecto/data_case.ex [hunk 1]
```diff
@@ -44,10 +44,11 @@ defmodule <%= app_module %>.DataCase do
...
   """
   def errors_on(changeset, field) do
-    for {message, opts} <- Keyword.get_values(changeset.errors, field) do
+    Ecto.Changeset.traverse_errors(changeset, fn {message, opts} ->
       Enum.reduce(opts, message, fn {key, value}, acc ->
         String.replace(acc, "%{#{key}}", to_string(value))
       end)
-    end
+    end)
+    |> Map.get(field, [])
```
Comment (josevalim) [inline installer/templates/phx_ecto/data_case.ex]: Credo has a check that does not allow a single pipe. So we should rather pipeline the whole way, starting with changeset or break it apart using a variable.

Comment (josevalim) [inline installer/templates/phx_ecto/data_case.ex]: Actually, let's go with `errors_on/1` that you have proposed. This way we get rid of the pipe and it is slightly more general purpose.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**